### PR TITLE
[Merged by Bors] - feat: port algebra.euclidean_domain.instances

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
Tracking mathlib commit: [cf9386b56953fb40904843af98b7a80757bbe7f9](https://github.com/leanprover-community/mathlib/commit/cf9386b56953fb40904843af98b7a80757bbe7f9)

It turns out the `Int` instance doesn't need `Data.Int.Order.Basic` because the dependencies moved to Std. Still looking at the `Field` instance

### CODE REVIEW

File: Mathlib/Algebra/EuclideanDomain/Instances.lean [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import Mathlib.Algebra.GroupWithZero.Units.Lemmas
+import Mathlib.Algebra.Order.Ring.Lemmas
+import Mathlib.Init.Data.Int.Order
+import Mathlib.Data.Int.Basic
+import Mathlib.Data.Nat.Order.Basic
+
+/-!
+# Instances for Euclidean domains
+* `int.euclidean_domain`: shows that `ℤ` is a Euclidean domain.
+* `field.to_euclidean_domain`: shows that any field is a Euclidean domain.
```
Comment (semorrison) [inline Mathlib/Algebra/EuclideanDomain/Instances.lean:17]:
```suggestion
* `Int.euclideanDomain`: shows that `ℤ` is a Euclidean domain.
* `Field.toEuclideanDomain`: shows that any field is a Euclidean domain.
```

**Comment by semorrison** [comment]
bors d+

Happy for you to merge, or mark for review again if you want to discuss the slowness of the `Field.toEuclideanDomain` declaration.

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
@Ruben-VandeVelde, could you take a look at this again?

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**